decidim-proposals/app/events/decidim/proposals/proposal_endorsed_event.rb

```diff
@@ -0,0 +1,49 @@
+# frozen-string_literal: true
+
+module Decidim
+  module Proposals
+    class ProposalEndorsedEvent < Decidim::Events::BaseEvent
```

Comment:
I'd suggest using `SimpleEvent` (I added it so it's easier to add new events), you could do something like this:

```ruby

i18n_attributes :endorser_nickname, :endorser_name, :endorser_path

delegate :nickname, :name, to :endorser, prefix: true

def endorser_path
  endorser.profile_path
end
```

And only keep `def endorser` you can remove all the other code.

Reply:
done